wordpress-mobile/WordPress-iOS
# Reading Preferences: Apply display setting to comments and related posts section

Part of #22925 

As titled, this applies the reading preferences to the comments and related posts sections. In addition, this PR also prevents the related posts from re-fetching content when the section is refreshed, preventing the content from changing every time the user tweaks the preferences.

Some previews:

<table>
<tr>
<td><img width="392" alt="comments-default" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/b228b23d-1a14-4bce-98d6-94af98b46a7a"></td>
<td><img width="386" alt="comments-oled" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/06e566ec-cb25-4d5a-b62b-9ded54dabf87"></td>
<td><img width="389" alt="comments-soft" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/8cc22364-69bb-4482-94b9-e0329c23189a"></td>
</tr>
<tr>
<td><img width="389" alt="comments-evening" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/255e0110-42e2-4b6c-a519-c319611c2d14"></td>
<td><img width="392" alt="comments-sepia" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/e3807711-1dde-4de5-900f-bc8eb7fdffa6"></td>
<td><img width="393" alt="Screenshot 2024-04-01 at 17 40 10" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/f5d86faf-138a-4cee-8459-b28d3c9c4be2"></td>
</tr>
</table>

Related posts:

<table>
<tr>
<td><img width="389" alt="related-default" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/225ecedf-c6f5-4464-aad2-840743bd58d1"></td>
<td><img width="386" alt="related-soft" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/c96fe415-407e-47ad-8547-7384a4040e43"></td>
<td><img width="388" alt="related-sepia" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/026656ed-4ff2-4f79-9b8d-f6d60229c469"></td>
</tr>
<tr>
<td><img width="387" alt="related-evening" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/b9315f2f-c48c-46b8-8826-49a46daddfaa"></td>
<td><img width="386" alt="related-oled" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/e57e0e8a-25c6-413b-b598-4b53e0fe584e"></td>
</tr>
</table>

## To test

- Launch the Jetpack app.
- Enable the `Reader Customization` feature flag.
- Navigate to the Reader and select any post with comments.
- Scroll down to the comments section.
- Customize the reading preferences.
- 🔎 Verify that the comments section is updated to match the selected preferences.
- Scroll down to the Related Posts section.
- 🔎 Verify that the related posts section is updated to match the selected preferences.

### Posts with no comments

- Navigate back to the Reader stream.
- Select any post that has no comments.
- Scroll down to the comments section.
- 🔎 Verify that the 'No comments yet' text is displayed and styled according to reading preferences. 

## Regression Notes
1. Potential unintended areas of impact
    - This modifies the `CommentContentTableViewCell`, which is used in site comments, notification comments, and comment thread.
    - This modifies the `ReaderDetailViewController` to only fetch the related posts data once. Previously, the related posts data is refreshed every time the web view that displays the post content is loaded.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

4. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:


Testing checklist:
